packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -107,6 +109,9 @@ class AndroidView extends StatefulWidget {
   /// {@endtemplate}
   final TextDirection layoutDirection;
 
+  /// How to compose the Android view.
+  final AndroidViewCompositeMode compositeMode;
```

Comment:
What happens if the widget is being rebuilt with a different surface type?

Reply:
Assuming we don't leak state in the external view embedder, or in the Android embedding, that would be the equivalent of detaching the previous view, and attaching a new one.